core/src/com/unciv/logic/automation/SpecificUnitAutomation.kt

```diff
@@ -420,4 +500,11 @@ object SpecificUnitAutomation {
         
         UnitActions.getEnhanceReligionAction(unit)()
     }
+
+    private fun doReligiousAction(unit: MapUnit, destination: TileInfo){
+        val actionList: java.util.ArrayList<UnitAction> = ArrayList()
+        UnitActions.addActionsWithLimitedUses(unit, actionList, destination)
+        if (actionList.firstOrNull()?.action?.invoke() == null) return
```

Comment:
It looks like you're actually invoking the action *twice* upon success!

Reply:
that was stupid of me. Fixed

Reply:
Looks like you still call the invoke action twice?